### ISSUE TITLE
BUG-report: Distance computation fails for valid geo. coordinates!

### ISSUE THREAD

**Issue by mkeller-upb** [geopy/geopy#30]
REWRITTEN 06.12.2013 (simpler description, improved mini-example, bugfix in mini-example)
UPDATED 31.12.2013 (revisit after changes in version 0.96.3, improved mini-example, used version 0.97)

For a couple of valid geo-coordinates distance-computations raise exception. This seems to me like a bug.

In the provided example code below, we have two lists of pairs of coordinates: list "a" geopy.distance.ELLIPSOIDS fails and list "b" fails both.

See my mini-example:

``` python
import pprint
import geopy
import geopy.distance


# format: list of point pairs
#   each point is specified as (lat, long)
#
# lat n/s: [90, -90]
# long w/e: [180, -180]
#
a = [[(6.5, -4.5),(-5.5, 176.5)],
[(37.5, 121.5),(-37.5, -58.5)],
[(-11.5, -77.5),(10.5, 103.5)],
[(-33.5, -58.5),(33.5, 121.5)],
[(6.5, -55.5),(-7.5, 123.5)],
[(31.5, 120.5),(-31.5, -59.5)],
[(32.5, 121.5),(-32.5, -58.5)],
[(-7.5, -79.5),(8.5, 99.5)],
[(-12.5, -171.5),(13.5, 7.5)],
[(-4.5, 106.5),(5.5, -74.5)]]

b = [[(-5.5, -79.5),(5.5, 100.5)],
[(-2.5, -59.5),(2.5, 120.5)],
[(5.5, -74.5),(-5.5, 105.5)],
[(-2.5, -52.5),(2.5, 127.5)],
[(2.5, 100.5),(-2.5, -79.5)],
[(-5.5, 106.5),(5.5, -73.5)],
[(-2.5, 104.5),(2.5, -75.5)],
[(-5.5, 116.5),(5.5, -63.5)],
[(-5.5, 121.5),(5.5, -58.5)]]

distance_variants = geopy.distance.ELLIPSOIDS.keys() + ['great_circle']


def check(points, variants):
    """ first checks geopy.Point conversion and then different variants of distance computation"""
    (lat1, long1), (lat2, long2) = points
    p1 = geopy.Point(lat1, long1)
    p2 = geopy.Point(lat2, long2)

    testlat1, testlong1, _ = p1
    testlat2, testlong2, _ = p2

    if (testlat1, testlong1) != (lat1, long1):
        print "ERROR: (%s, %s) = geopy.Point(%s,%s)" % (testlat1, testlong1, lat1, long1)
        return

    if (testlat2, testlong2) != (lat2, long2):
        print "ERROR: (%s, %s) = geopy.Point(%s,%s)" % (testlat1, testlong1, lat1, long1)
        return

    print "Distance in km between %r <-> %r" % (p1, p2)
    for dv in variants:
        res = ""
        if dv == "great_circle":
            try:
                res = geopy.distance.great_circle(p1,p2).km
            except Exception, e:
                res = "ERROR: " + str(e)
        else:
            for iterations in (20, 40, 80, 160, 320, 640, 1280, 2560, 5120, 10240, 20480, 40960, 81920, 163840, 327680):
                try:
                    res = geopy.distance.distance(p1,p2,ellipsoid=dv,iterations=iterations).km
                    res = "%s km  (iterations=%s)" % (res, iterations)
                    break
                except Exception, e:
                    res = "ERROR: %s  (iterations=%s)" % (e, iterations)
        print "%20s : %s" % (dv, res)


print "ERRORS in geopy.distance.ELLIPSOIDS (distance in km)"
for item in a:
    check(item, distance_variants)

print "ERRORS in geopy.distance.great_circle (distance in km)"
for item in b:
    check(item, distance_variants)
```

results in:
```ERRORS in geopy.distance.ELLIPSOIDS (distance in km)
Distance in km between Point(6.5, -4.5, 0.0) <-> Point(-5.5, 176.5, 0.0)
       Clarke (1880) : 19860.699201 km  (iterations=40)
              GRS-67 : 19860.7928533 km  (iterations=20)
         Airy (1830) : 19859.0166797 km  (iterations=20)
              WGS-84 : 19860.7217849 km  (iterations=20)
              GRS-80 : 19860.7217848 km  (iterations=20)
           Intl 1924 : 19861.4071186 km  (iterations=20)
        great_circle : 19863.8595647
Distance in km between Point(37.5, 121.5, 0.0) <-> Point(-37.5, -58.5, 0.0)
       Clarke (1880) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-67 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
         Airy (1830) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              WGS-84 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-80 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
           Intl 1924 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
        great_circle : 20020.7259548
...
ERRORS in geopy.distance.great_circle (distance in km)
Distance in km between Point(-5.5, -79.5, 0.0) <-> Point(5.5, 100.5, 0.0)
       Clarke (1880) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-67 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
         Airy (1830) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              WGS-84 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-80 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
           Intl 1924 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
        great_circle : 20020.7259548
...

```
(CHANGED: Test script now dry increasing number of iterations to obtain solution for a distance.)

(MacOSX, Version geopy-0.97-py2.7.egg)

Observations:
- Great circle distance: Math-exception is fixed; all coordinates in example work
- Ellipsoid computation: Increasing the iterations only work for Clarke-Approximation (for the example coordinates).
  A dramatic increase of the number of iterations doesn't really help. This hints (at least for me) towards a floating point issue.
- Used coordinates are valid and distance between them should work.
(Updated bug summary)

Please help on this topic.
Thank you for your help so far.
```

**Comment by unnir** [comment]
any updates?

**Comment by bluetrickpt** [comment]
Just tested these use cases and the bug is still there.
I am using geopy version 1.11.0 in linux (fedora 27).

**Comment by KostyaEsmukov** [comment]
Thank you @mkeller-upb for providing this very useful snippet!

#144 has introduced a new algorithm for distance computations which is not affected by this issue of Vincenty being unable to converge for antipodal points. It's going to be released soon in `1.13.0` version of geopy. The default `distance` alias has also been changed to `geodesic` from `vincenty`.

<details/>
  <summary>Snippet output for geopy.distance.distance (currently geopy.distance.geodesic)</summary>

```
ERRORS in geopy.distance.ELLIPSOIDS (distance in km)
Distance in km between Point(6.5, -4.5, 0.0) <-> Point(-5.5, 176.5, 0.0)
[...]/geopy/geopy/distance.py:354: UserWarning: Ignoring unused `iterations` kwarg for geodesic distance.
  'distance.', UserWarning)
       Clarke (1880) : 19860.699201 km  (iterations=20)
              GRS-67 : 19860.7928537 km  (iterations=20)
         Airy (1830) : 19859.0166801 km  (iterations=20)
              WGS-84 : 19860.721785 km  (iterations=20)
              GRS-80 : 19860.7217849 km  (iterations=20)
           Intl 1924 : 19861.4071188 km  (iterations=20)
        great_circle : 19858.2926426
Distance in km between Point(37.5, 121.5, 0.0) <-> Point(-37.5, -58.5, 0.0)
       Clarke (1880) : 20003.7351033 km  (iterations=20)
              GRS-67 : 20004.0027813 km  (iterations=20)
         Airy (1830) : 20002.2521614 km  (iterations=20)
              WGS-84 : 20003.9314586 km  (iterations=20)
              GRS-80 : 20003.9314585 km  (iterations=20)
           Intl 1924 : 20004.576598 km  (iterations=20)
        great_circle : 20015.1150704
Distance in km between Point(-11.5, -77.5, 0.0) <-> Point(10.5, 103.5, 0.0)
       Clarke (1880) : 19861.1992485 km  (iterations=20)
              GRS-67 : 19861.2986958 km  (iterations=20)
         Airy (1830) : 19859.5237614 km  (iterations=20)
              WGS-84 : 19861.2276339 km  (iterations=20)
              GRS-80 : 19861.2276338 km  (iterations=20)
           Intl 1924 : 19861.9114752 km  (iterations=20)
        great_circle : 19859.3004898
Distance in km between Point(-33.5, -58.5, 0.0) <-> Point(33.5, 121.5, 0.0)
       Clarke (1880) : 20003.7351033 km  (iterations=20)
              GRS-67 : 20004.0027813 km  (iterations=20)
         Airy (1830) : 20002.2521614 km  (iterations=20)
              WGS-84 : 20003.9314586 km  (iterations=20)
              GRS-80 : 20003.9314585 km  (iterations=20)
           Intl 1924 : 20004.576598 km  (iterations=20)
        great_circle : 20015.1150704
Distance in km between Point(6.5, -55.5, 0.0) <-> Point(-7.5, 123.5, 0.0)
       Clarke (1880) : 19860.775629 km  (iterations=20)
              GRS-67 : 19860.8701745 km  (iterations=20)
         Airy (1830) : 19859.0941919 km  (iterations=20)
              WGS-84 : 19860.7991068 km  (iterations=20)
              GRS-80 : 19860.7991067 km  (iterations=20)
           Intl 1924 : 19861.4842107 km  (iterations=20)
        great_circle : 19858.447497
Distance in km between Point(31.5, 120.5, 0.0) <-> Point(-31.5, -59.5, 0.0)
       Clarke (1880) : 20003.7351033 km  (iterations=20)
              GRS-67 : 20004.0027813 km  (iterations=20)
         Airy (1830) : 20002.2521614 km  (iterations=20)
              WGS-84 : 20003.9314586 km  (iterations=20)
              GRS-80 : 20003.9314585 km  (iterations=20)
           Intl 1924 : 20004.576598 km  (iterations=20)
        great_circle : 20015.1150704
Distance in km between Point(32.5, 121.5, 0.0) <-> Point(-32.5, -58.5, 0.0)
       Clarke (1880) : 20003.7351033 km  (iterations=20)
              GRS-67 : 20004.0027813 km  (iterations=20)
         Airy (1830) : 20002.2521614 km  (iterations=20)
              WGS-84 : 20003.9314586 km  (iterations=20)
              GRS-80 : 20003.9314585 km  (iterations=20)
           Intl 1924 : 20004.576598 km  (iterations=20)
        great_circle : 20015.1150704
Distance in km between Point(-7.5, -79.5, 0.0) <-> Point(8.5, 99.5, 0.0)
       Clarke (1880) : 19860.8638383 km  (iterations=20)
              GRS-67 : 19860.959411 km  (iterations=20)
         Airy (1830) : 19859.1836481 km  (iterations=20)
              WGS-84 : 19860.8883445 km  (iterations=20)
              GRS-80 : 19860.8883444 km  (iterations=20)
           Intl 1924 : 19861.5731838 km  (iterations=20)
        great_circle : 19858.6258554
Distance in km between Point(-12.5, -171.5, 0.0) <-> Point(13.5, 7.5, 0.0)
       Clarke (1880) : 19861.4819428 km  (iterations=20)
              GRS-67 : 19861.5846175 km  (iterations=20)
         Airy (1830) : 19859.8103729 km  (iterations=20)
              WGS-84 : 19861.5135594 km  (iterations=20)
              GRS-80 : 19861.5135593 km  (iterations=20)
           Intl 1924 : 19862.1965695 km  (iterations=20)
        great_circle : 19859.8647569
Distance in km between Point(-4.5, 106.5, 0.0) <-> Point(5.5, -74.5, 0.0)
       Clarke (1880) : 19860.634546 km  (iterations=20)
              GRS-67 : 19860.7274415 km  (iterations=20)
         Airy (1830) : 19858.9511059 km  (iterations=20)
              WGS-84 : 19860.6563719 km  (iterations=20)
              GRS-80 : 19860.6563717 km  (iterations=20)
           Intl 1924 : 19861.3419008 km  (iterations=20)
        great_circle : 19858.161411
ERRORS in geopy.distance.great_circle (distance in km)
Distance in km between Point(-5.5, -79.5, 0.0) <-> Point(5.5, 100.5, 0.0)
       Clarke (1880) : 20003.7351033 km  (iterations=20)
              GRS-67 : 20004.0027813 km  (iterations=20)
         Airy (1830) : 20002.2521614 km  (iterations=20)
              WGS-84 : 20003.9314586 km  (iterations=20)
              GRS-80 : 20003.9314585 km  (iterations=20)
           Intl 1924 : 20004.576598 km  (iterations=20)
        great_circle : 20015.1150704
Distance in km between Point(-2.5, -59.5, 0.0) <-> Point(2.5, 120.5, 0.0)
       Clarke (1880) : 20003.7351033 km  (iterations=20)
              GRS-67 : 20004.0027813 km  (iterations=20)
         Airy (1830) : 20002.2521614 km  (iterations=20)
              WGS-84 : 20003.9314586 km  (iterations=20)
              GRS-80 : 20003.9314585 km  (iterations=20)
           Intl 1924 : 20004.576598 km  (iterations=20)
        great_circle : 20015.1150704
Distance in km between Point(5.5, -74.5, 0.0) <-> Point(-5.5, 105.5, 0.0)
       Clarke (1880) : 20003.7351033 km  (iterations=20)
              GRS-67 : 20004.0027813 km  (iterations=20)
         Airy (1830) : 20002.2521614 km  (iterations=20)
              WGS-84 : 20003.9314586 km  (iterations=20)
              GRS-80 : 20003.9314585 km  (iterations=20)
           Intl 1924 : 20004.576598 km  (iterations=20)
        great_circle : 20015.1150704
Distance in km between Point(-2.5, -52.5, 0.0) <-> Point(2.5, 127.5, 0.0)
       Clarke (1880) : 20003.7351033 km  (iterations=20)
              GRS-67 : 20004.0027813 km  (iterations=20)
         Airy (1830) : 20002.2521614 km  (iterations=20)
              WGS-84 : 20003.9314586 km  (iterations=20)
              GRS-80 : 20003.9314585 km  (iterations=20)
           Intl 1924 : 20004.576598 km  (iterations=20)
        great_circle : 20015.1150704
Distance in km between Point(2.5, 100.5, 0.0) <-> Point(-2.5, -79.5, 0.0)
       Clarke (1880) : 20003.7351033 km  (iterations=20)
              GRS-67 : 20004.0027813 km  (iterations=20)
         Airy (1830) : 20002.2521614 km  (iterations=20)
              WGS-84 : 20003.9314586 km  (iterations=20)
              GRS-80 : 20003.9314585 km  (iterations=20)
           Intl 1924 : 20004.576598 km  (iterations=20)
        great_circle : 20015.1150704
Distance in km between Point(-5.5, 106.5, 0.0) <-> Point(5.5, -73.5, 0.0)
       Clarke (1880) : 20003.7351033 km  (iterations=20)
              GRS-67 : 20004.0027813 km  (iterations=20)
         Airy (1830) : 20002.2521614 km  (iterations=20)
              WGS-84 : 20003.9314586 km  (iterations=20)
              GRS-80 : 20003.9314585 km  (iterations=20)
           Intl 1924 : 20004.576598 km  (iterations=20)
        great_circle : 20015.1150704
Distance in km between Point(-2.5, 104.5, 0.0) <-> Point(2.5, -75.5, 0.0)
       Clarke (1880) : 20003.7351033 km  (iterations=20)
              GRS-67 : 20004.0027813 km  (iterations=20)
         Airy (1830) : 20002.2521614 km  (iterations=20)
              WGS-84 : 20003.9314586 km  (iterations=20)
              GRS-80 : 20003.9314585 km  (iterations=20)
           Intl 1924 : 20004.576598 km  (iterations=20)
        great_circle : 20015.1150704
Distance in km between Point(-5.5, 116.5, 0.0) <-> Point(5.5, -63.5, 0.0)
       Clarke (1880) : 20003.7351033 km  (iterations=20)
              GRS-67 : 20004.0027813 km  (iterations=20)
         Airy (1830) : 20002.2521614 km  (iterations=20)
              WGS-84 : 20003.9314586 km  (iterations=20)
              GRS-80 : 20003.9314585 km  (iterations=20)
           Intl 1924 : 20004.576598 km  (iterations=20)
        great_circle : 20015.1150704
Distance in km between Point(-5.5, 121.5, 0.0) <-> Point(5.5, -58.5, 0.0)
       Clarke (1880) : 20003.7351033 km  (iterations=20)
              GRS-67 : 20004.0027813 km  (iterations=20)
         Airy (1830) : 20002.2521614 km  (iterations=20)
              WGS-84 : 20003.9314586 km  (iterations=20)
              GRS-80 : 20003.9314585 km  (iterations=20)
           Intl 1924 : 20004.576598 km  (iterations=20)
        great_circle : 20015.1150704
```

</details>


<details/>
  <summary>Snippet output for geopy.distance.vincenty</summary>

```
ERRORS in geopy.distance.ELLIPSOIDS (distance in km)
Distance in km between Point(6.5, -4.5, 0.0) <-> Point(-5.5, 176.5, 0.0)
       Clarke (1880) : 19860.699201 km  (iterations=40)
              GRS-67 : 19860.7928533 km  (iterations=20)
         Airy (1830) : 19859.0166797 km  (iterations=20)
              WGS-84 : 19860.7217849 km  (iterations=20)
              GRS-80 : 19860.7217848 km  (iterations=20)
           Intl 1924 : 19861.4071186 km  (iterations=20)
        great_circle : 19858.2926426
Distance in km between Point(37.5, 121.5, 0.0) <-> Point(-37.5, -58.5, 0.0)
       Clarke (1880) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-67 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
         Airy (1830) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              WGS-84 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-80 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
           Intl 1924 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
        great_circle : 20015.1150704
Distance in km between Point(-11.5, -77.5, 0.0) <-> Point(10.5, 103.5, 0.0)
       Clarke (1880) : 19861.1992486 km  (iterations=20)
              GRS-67 : 19861.2986954 km  (iterations=20)
         Airy (1830) : 19859.523761 km  (iterations=20)
              WGS-84 : 19861.2276339 km  (iterations=20)
              GRS-80 : 19861.2276338 km  (iterations=20)
           Intl 1924 : 19861.9114749 km  (iterations=20)
        great_circle : 19859.3004898
Distance in km between Point(-33.5, -58.5, 0.0) <-> Point(33.5, 121.5, 0.0)
       Clarke (1880) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-67 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
         Airy (1830) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              WGS-84 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-80 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
           Intl 1924 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
        great_circle : 20015.1150704
Distance in km between Point(6.5, -55.5, 0.0) <-> Point(-7.5, 123.5, 0.0)
       Clarke (1880) : 19860.775629 km  (iterations=40)
              GRS-67 : 19860.8701741 km  (iterations=20)
         Airy (1830) : 19859.0941914 km  (iterations=20)
              WGS-84 : 19860.7991068 km  (iterations=20)
              GRS-80 : 19860.7991067 km  (iterations=20)
           Intl 1924 : 19861.4842105 km  (iterations=20)
        great_circle : 19858.447497
Distance in km between Point(31.5, 120.5, 0.0) <-> Point(-31.5, -59.5, 0.0)
       Clarke (1880) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-67 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
         Airy (1830) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              WGS-84 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-80 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
           Intl 1924 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
        great_circle : 20015.1150704
Distance in km between Point(32.5, 121.5, 0.0) <-> Point(-32.5, -58.5, 0.0)
       Clarke (1880) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-67 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
         Airy (1830) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              WGS-84 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-80 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
           Intl 1924 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
        great_circle : 20015.1150704
Distance in km between Point(-7.5, -79.5, 0.0) <-> Point(8.5, 99.5, 0.0)
       Clarke (1880) : 19860.8638383 km  (iterations=40)
              GRS-67 : 19860.9594106 km  (iterations=20)
         Airy (1830) : 19859.1836477 km  (iterations=20)
              WGS-84 : 19860.8883445 km  (iterations=20)
              GRS-80 : 19860.8883444 km  (iterations=20)
           Intl 1924 : 19861.5731836 km  (iterations=20)
        great_circle : 19858.6258554
Distance in km between Point(-12.5, -171.5, 0.0) <-> Point(13.5, 7.5, 0.0)
       Clarke (1880) : 19861.4819429 km  (iterations=20)
              GRS-67 : 19861.5846171 km  (iterations=20)
         Airy (1830) : 19859.8103724 km  (iterations=20)
              WGS-84 : 19861.5135594 km  (iterations=20)
              GRS-80 : 19861.5135593 km  (iterations=20)
           Intl 1924 : 19862.1965693 km  (iterations=20)
        great_circle : 19859.8647569
Distance in km between Point(-4.5, 106.5, 0.0) <-> Point(5.5, -74.5, 0.0)
       Clarke (1880) : 19860.634546 km  (iterations=40)
              GRS-67 : 19860.7274411 km  (iterations=20)
         Airy (1830) : 19858.9511054 km  (iterations=20)
              WGS-84 : 19860.6563718 km  (iterations=20)
              GRS-80 : 19860.6563717 km  (iterations=20)
           Intl 1924 : 19861.3419006 km  (iterations=40)
        great_circle : 19858.161411
ERRORS in geopy.distance.great_circle (distance in km)
Distance in km between Point(-5.5, -79.5, 0.0) <-> Point(5.5, 100.5, 0.0)
       Clarke (1880) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-67 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
         Airy (1830) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              WGS-84 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-80 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
           Intl 1924 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
        great_circle : 20015.1150704
Distance in km between Point(-2.5, -59.5, 0.0) <-> Point(2.5, 120.5, 0.0)
       Clarke (1880) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-67 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
         Airy (1830) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              WGS-84 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-80 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
           Intl 1924 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
        great_circle : 20015.1150704
Distance in km between Point(5.5, -74.5, 0.0) <-> Point(-5.5, 105.5, 0.0)
       Clarke (1880) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-67 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
         Airy (1830) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              WGS-84 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-80 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
           Intl 1924 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
        great_circle : 20015.1150704
Distance in km between Point(-2.5, -52.5, 0.0) <-> Point(2.5, 127.5, 0.0)
       Clarke (1880) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-67 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
         Airy (1830) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              WGS-84 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-80 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
           Intl 1924 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
        great_circle : 20015.1150704
Distance in km between Point(2.5, 100.5, 0.0) <-> Point(-2.5, -79.5, 0.0)
       Clarke (1880) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-67 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
         Airy (1830) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              WGS-84 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-80 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
           Intl 1924 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
        great_circle : 20015.1150704
Distance in km between Point(-5.5, 106.5, 0.0) <-> Point(5.5, -73.5, 0.0)
       Clarke (1880) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-67 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
         Airy (1830) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              WGS-84 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-80 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
           Intl 1924 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
        great_circle : 20015.1150704
Distance in km between Point(-2.5, 104.5, 0.0) <-> Point(2.5, -75.5, 0.0)
       Clarke (1880) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-67 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
         Airy (1830) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              WGS-84 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-80 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
           Intl 1924 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
        great_circle : 20015.1150704
Distance in km between Point(-5.5, 116.5, 0.0) <-> Point(5.5, -63.5, 0.0)
       Clarke (1880) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-67 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
         Airy (1830) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              WGS-84 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-80 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
           Intl 1924 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
        great_circle : 20015.1150704
Distance in km between Point(-5.5, 121.5, 0.0) <-> Point(5.5, -58.5, 0.0)
       Clarke (1880) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-67 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
         Airy (1830) : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              WGS-84 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
              GRS-80 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
           Intl 1924 : ERROR: Vincenty formula failed to converge!  (iterations=327680)
        great_circle : 20015.1150704
```

</details>
